runtime/src/main/java/org/capnproto/AnyList.java

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) 2013-2014 Sandstorm Development Group, Inc. and contributors
+// Licensed under the MIT License:
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package org.capnproto;
+
+public final class AnyList {
+
+    public static final class Factory extends ListFactory<Builder, Reader> {
+        Factory() {
+            super(ElementSize.VOID);
```

Comment:
Does this end up giving a wrong value for the `expectedElementSize` parameter of `WireHelpers.readListPointer()`? Maybe we need to add a `checkElementSize` parameter, like in capnproto-c++? https://github.com/capnproto/capnproto/blob/master/c%2B%2B/src/capnp/layout.c%2B%2B#L2237

Reply:
That probably explains why my round-trip test for AnyList fails!

Reply:
Am I right in thinking that the actual element size has to be written out when initListPointer is called?
In that case, I think I need to allow the user to pass the appropriate factory to the generated accessor when initiialising the list for building.

Reply:
Yes, I agree. For a field
```capnp
  foo @1 :AnyList
```
I think we should generate a method like `public <B> B initFooAs(ListFactory<B, ?> factory, int size)`.
(capnproto-c++ does something similar)